codebase-pipelines/codebuild.tf

```diff
@@ -0,0 +1,115 @@
+data "aws_codestarconnections_connection" "github_codestar_connection" {
+  name = var.application
+}
+
+resource "aws_codebuild_project" "codebase_image_build" {
+  name          = "${var.application}-${var.codebase}-codebase-image-build"
+  description   = "Publish images on push to ${var.repository}"
+  build_timeout = 30
+  service_role  = aws_iam_role.codebase_image_build.arn
+  badge_enabled = true
+
+  artifacts {
+    type = "NO_ARTIFACTS"
+  }
+
+  cache {
+    type  = "LOCAL"
+    modes = ["LOCAL_DOCKER_LAYER_CACHE"]
+  }
+
+  environment {
+    compute_type = "BUILD_GENERAL1_SMALL"
+    image        = "public.ecr.aws/uktrade/ci-image-builder:tag-latest"
+    type         = "LINUX_CONTAINER"
+
+    environment_variable {
+      name  = "AWS_ACCOUNT_ID"
+      value = data.aws_caller_identity.current.account_id
+    }
+
+    environment_variable {
+      name  = "ECR_REPOSITORY"
+      value = local.ecr_name
+    }
+
+    environment_variable {
+      name  = "CODESTAR_CONNECTION_ARN"
+      value = data.aws_codestarconnections_connection.github_codestar_connection.arn
+    }
+
+    dynamic "environment_variable" {
+      for_each = var.additional_ecr_repository != null ? [1] : []
+      content {
+        name  = "ADDITIONAL_ECR_REPOSITORY"
+        value = var.additional_ecr_repository
+      }
+    }
+  }
+
+  logs_config {
+    cloudwatch_logs {
+      group_name  = aws_cloudwatch_log_group.codebase_image_build.name
+      stream_name = aws_cloudwatch_log_stream.codebase_image_build.name
+    }
+  }
+
+  source {
+    type            = "GITHUB"
+    buildspec       = file("${path.module}/buildspec.yml")
+    location        = "https://github.com/${var.repository}.git"
+    git_clone_depth = 0
+    git_submodules_config {
+      fetch_submodules = false
+    }
+  }
+
+  tags = local.tags
+}
+
+resource "aws_cloudwatch_log_group" "codebase_image_build" {
+  # checkov:skip=CKV_AWS_338:Retains logs for 3 months instead of 1 year
+  # checkov:skip=CKV_AWS_158:To be reworked
```

Comment:
Can we include words that indicate the nature of the suppression without us having to go and look it up?